src/initializers/rabbitmq/index.ts

```diff
@@ -106,3 +111,22 @@ export const close = async () => {
     healthy = false;
   }
 };
+
+function methodCreator(originalSendToQueue, paramNum, config) {
+  return function method(...args: any) {
+    let properties = args[paramNum - 1];
+    if (!properties) {
+      properties = {};
+      args[paramNum - 1] = properties;
+    }
+
+    const traceHeaderName = config.traceHeaderName.toLowerCase();
+    appendHeadersFromStore(properties, getRequestContext(), config);
+
+    if (!properties.correlationId && properties.headers && properties.headers[traceHeaderName]) {
+      properties.correlationId = properties.headers[traceHeaderName];
+    }
+
+    return originalSendToQueue.call(this, ...args);
+  };
+}
```

Comment:
❓ Can you please help me understand what this method and paramNum does?

Reply:
It gets a method and updates it's properties from the requestedContext.

The paramNum is the place in the argument list that the properties is located because it is different for exchanges and queues. See it's usage here https://github.com/Workable/orka/pull/339/files/3b40e032b2f42a05852ff966de75eb69e72c4083#diff-b894bbadeee79e29ba96f8b22f9eb10bacaf29211f93e3e96c33b8936315328aR44-R47

Reply:
Maybe sth like `propertiesObjectIndex` sounds clear to me after your explanation. Wdyt?

Reply:
propertiesParamIndex or propertiesArgIndex maybe?

Reply:
updated 9d886f8883149b2c4ea10ed735179a5a5753178e